clients/pkg/promtail/targets/gcppush/target_test.go

```diff
@@ -0,0 +1,300 @@
+package gcppush
+
+import (
+	"flag"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/model/relabel"
+	"github.com/stretchr/testify/require"
+	"github.com/weaveworks/common/server"
+
+	lokiClient "github.com/grafana/loki/clients/pkg/promtail/client"
+	"github.com/grafana/loki/clients/pkg/promtail/client/fake"
+	"github.com/grafana/loki/clients/pkg/promtail/scrapeconfig"
+)
+
+const localhost = "127.0.0.1"
+
+const expectedMessageData = `{
+  "severity": "DEBUG",
+  "textPayload": "Function execution took 1198 ms. Finished with status code: 200",
+  "trace": "projects/wired-height/traces/54aa8a20875253c6b47caa8bd28ad652"
+}`
+const testPayload = `
+{
+	"message": {
+		"attributes": {
+			"logging.googleapis.com/timestamp": "2022-07-25T22:19:09.903683708Z"
+		},
+		"data": "ewogICJzZXZlcml0eSI6ICJERUJVRyIsCiAgInRleHRQYXlsb2FkIjogIkZ1bmN0aW9uIGV4ZWN1dGlvbiB0b29rIDExOTggbXMuIEZpbmlzaGVkIHdpdGggc3RhdHVzIGNvZGU6IDIwMCIsCiAgInRyYWNlIjogInByb2plY3RzL3dpcmVkLWhlaWdodC90cmFjZXMvNTRhYThhMjA4NzUyNTNjNmI0N2NhYThiZDI4YWQ2NTIiCn0=",
+		"messageId": "5187581549398349",
+		"message_id": "5187581549398349",
+		"publishTime": "2022-07-25T22:19:15.56Z",
+		"publish_time": "2022-07-25T22:19:15.56Z"
+	},
+	"subscription": "projects/wired-height-350515/subscriptions/test"
+}`
```

Comment:
Sample payload obtained by setting up a dummy push subscription receiver that logs everything. The `message.data` was simplified, which is just decoded and forwarded to loki, and it can vary depending on what is forwarded from the subscribed topic.